integration-libs/opf/base/core/facade/opf-global-functions.service.ts

```diff
@@ -197,4 +221,58 @@ export class OpfGlobalFunctionsService implements OpfGlobalFunctionsFacade {
       });
     };
   }
+
+  protected registerSubmitCompleteRedirect(
```

Comment:
Could you consider splitting this method for a few smaller methods? It seems to be too long, complex and hard to read.

Reply:
done, created runSubmitComplete method that get called twice.